domain/domain.go

```diff
@@ -290,6 +290,10 @@ func (do *Domain) InfoSchema() infoschema.InfoSchema {
 
 // GetSnapshotInfoSchema gets a snapshot information schema.
 func (do *Domain) GetSnapshotInfoSchema(snapshotTS uint64) (infoschema.InfoSchema, error) {
+	// if the snapshotTS is new enough, we can get infoschema directly through sanpshotTS.
+	if is := do.infoCache.GetBySnapshotTS(snapshotTS); is != nil {
+		return is, nil
+	}
```

Comment:
I think this may cause the infoschema inconsistency in following case.

```sh
1. infoschema cache holding `t1` as maxUpdatedSnapshotTS and infoschema for `schema version` v1
2. ddl worker running ddl job with `t2` in txn and increased `schema version` to v2 while infoschema cache haven't been awared of this changing yet.
3. load infoschema with snapshotTS t3 (t3 > t2 > t1) and using infoschema with v1 schema version.
```

Reply:
`Infocache` is used everywhere, it is promised to have the latest infoschema for the current tidb instance(not necessarily the latest for all tidb instances).

Reply:
In my example, I think we don't need to guarantee that the query must use the schema version v2's infoschema if the changing is not related to the tables we visit.

But I think the following checking is ncessary:

1. check whether there exists any schema version changed bewtween `t1`(maxUpdatedSnapshotTS)  and `t3`(snapshotTS)
2. If exists, check whether the changing is related to the table we vistied.
3. If related, reloading the infoschema with `t3` snapshotTS to use the right infoschema.

Reply:
In my example, I think we don't need to guarantee that the query must use the schema version v2's infoschema if the changing is not related to the tables we visit.

But I think the following checking is necessary:

1. check whether there exists any schema version changed bewtween `t1`(maxUpdatedSnapshotTS)  and `t3`(snapshotTS)
2. If exists, check whether the changing is related to the table we vistied.
3. If related, reloading the infoschema with `t3` snapshotTS to use the right infoschema.